select_samples_grain_size.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.ndimage import center_of_mass
+
+
+def select_samples_grain_size(mask, bins, num_samples=1, setting='highest'):
+    # Find the sizes of the different particles
+    sizes = np.zeros(np.max(mask)+1)
+
+    for ii in range(np.max(mask)+1):
+        sizes[ii] = np.count_nonzero(mask==ii)
+
+    # Remove the zeroth element (the background)
+    sizes_new = sizes[1:]
+
+    # Bin these clusters by size
+    hist = np.histogram(sizes_new, bins=bins)
+    bin_boundaries = hist[1]
+
+    # Find the corresponding indices
+    size_args = []
+
+    for ii in range(bins):
+        bounds = (bin_boundaries[ii], bin_boundaries[ii+1])
+        bool_arr = np.array([sizes_new>=bounds[0], sizes_new<bounds[1]])
+        size_args.append(list(np.argwhere(bool_arr.all(0)).T[0]))
+
+    # Take samples from each bin
+    sample_list = []
+
+    for ii in range(bins):
```

Comment:
In general, we should be using useful loop variables. 

for example (I am not sure this actually means `bin_num`, so make it what it means):

```python
for bin_num in range(bins):
	...
```